pkg/controller/deployment/rolling.go

```diff
@@ -174,7 +173,7 @@ func (dc *DeploymentController) cleanupUnhealthyReplicas(ctx context.Context, ol
 			continue
 		}
 
-		scaledDownCount := int32(integer.IntMin(int(maxCleanupCount-totalScaledDown), int(*(targetRS.Spec.Replicas)-targetRS.Status.AvailableReplicas)))
+		scaledDownCount := int32(min(int(maxCleanupCount-totalScaledDown), int(*(targetRS.Spec.Replicas)-targetRS.Status.AvailableReplicas)))
```

Comment:
Perhaps we can omit unnecessary type conversions here.

Just like `scaledDownCount := min(maxCleanupCount-totalScaledDown, *targetRS.Spec.Replicas-targetRS.Status.AvailableReplicas)`

Reply:
Thanks for pointing this out.
Fixed.